test/scripts/hexo/load_theme_config.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const { join } = require('path');
+const { mkdirs, unlink, writeFile, rmdir } = require('hexo-fs');
+
+describe('Load alternate theme config', () => {
+  const Hexo = require('../../../lib/hexo');
+  const hexo = new Hexo(join(__dirname, 'config_test'), {silent: true});
+  const loadThemeConfig = require('../../../lib/hexo/load_theme_config');
+
+  hexo.env.init = true;
+
+  before(() => mkdirs(hexo.base_dir).then(() => hexo.init()));
+
+  after(() => rmdir(hexo.base_dir));
+
+  beforeEach(() => {
+    hexo.config.theme_config = { foo: { bar: 'ahhhhhh' } };
+    hexo.config.theme = 'test_theme';
+  });
+
+  it('_config.[theme].yml does not exist', () => loadThemeConfig(hexo).then(() => {
+    hexo.config.theme_config = {};
+  }));
+
+  it('_config.[theme].yml exists', () => {
+    const configPath = join(hexo.base_dir, '_config.test_theme.yml');
+
+    return writeFile(configPath, 'bar: 1').then(() => loadThemeConfig(hexo)).then(() => {
+      hexo.config.theme_config.bar.should.eql(1);
+    }).finally(() => unlink(configPath));
+  });
+
+  it('_config.[theme].json exists', () => {
+    const configPath = join(hexo.base_dir, '_config.test_theme.json');
```

Comment:
Is this feature will allow other than `YAML` format?

Reply:
Since I have just port the feature from `load_config`. So if Hexo support `_config.json`, then `_config.test_theme.json` should be supported as well (That's why I include this in test cases)